scripts/create_newcase

```diff
@@ -640,6 +640,12 @@ sub _check_input_options
 	} else {
 	    die "ERROR: create_newcase must include the input argument, -res\n";
 	}
+	if($opts{mach_dir}){
+	    $mach_dir = $opts{mach_dir};
+	}else{
+	    $mach_dir => "$cimeroot/cime_config/${model}/machines",
```

Comment:
Hardcoding this is inconsistent with the desire to put model specific data into xml. The config object is available at this point, so this should be dirname($config->get('MACHINES_SPEC_FILE')). This is also duplicated below around line 869.